### ISSUE TITLE
media-feature-name-no-unknown

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#1698]
Valid:
```css
@media (max-width: 600px) {
}
```

Invalid:
```css
@media (unknown: 600px) {
```

**Comment by woble** [comment]
Should respect [CSS Media Queries Level 4](https://drafts.csswg.org/mediaqueries/#mq-range-context).

**Comment by evilebottnawi** [comment]
@woble yep

**Comment by jeddy3** [comment]
> Rule idea `media-feature-name-no-unknown`

SGTM. 

> Should respect CSS Media Queries Level 4 operators.

This would probably involve [writing a parser](https://github.com/dryoma/postcss-media-query-parser/issues/1#issuecomment-233054873) for Media Features in a Range Context.

@stylelint/core Any objections if I mark this up as "help wanted", and then the details around range contexts can be ironed out once someone has time to build this rule?

**Comment by davidtheclark** [comment]
Sure.

**Comment by jeddy3** [comment]
- Name: `media-feature-name-no-unknown`
- Primary option: `true`
- Secondary options: `ignoreMediaFeatureNames: []` (user defined array)
- Message: "Unexpected unknown media feature name"
- Section: "Media feature"
